### PR TITLE
[Example] Add Automatic Speech Recognition Sample

### PR DESCRIPTION
**Pull request by miaobin**
Implement ASR example with @mingmingtasd  and get the same result as native C++ sample.
@belem @BruceDai @fujunwei  PTAL.

Thanks!

### CODE REVIEW

File: examples/asr_sample/utils.js [hunk 1]
```diff
@@ -216,6 +226,83 @@ class Utils {
...
+        this.frameError.sumSquaredError += squared_error;
+        if (error > this.frameError.maxError) {
+          this.frameError.maxError = error;
+        }
+        this.frameError.sumRelError += rel_error;
+        this.frameError.sumSquaredRelError += squared_rel_error;
+        if (rel_error > this.frameError.maxRelError) {
+          this.frameError.maxRelError = rel_error;
+        }
+        if (error > this.frameError.threshold) {
```
Comment (fujunwei) [inline examples/asr_sample/utils.js]: OpenVINO GNA use Int8/Int16 to inference the model, maybe the threshold should be 1 for GNA, the below are result run by native speech sample
```
         max error: 0.705184
         avg error: 0.0448388
         avg rms error: 0.0574098
         stdev error: 0.0371649

```

Comment (mingmingtasd) [reply examples/asr_sample/utils.js]: Now just defined the value of threshold according to the [native definition](https://github.com/opencv/dldt/blob/2020/inference-engine/samples/speech_sample/main.cpp#L32). And I checked that when computing errors, native code uses float32 [here](https://github.com/opencv/dldt/blob/2020/inference-engine/samples/speech_sample/main.cpp#L224).

Comment (fujunwei) [reply examples/asr_sample/utils.js]: > Now just defined the value of threshold according to the native definition. 

The native sample does't use the error num.

> And I checked that when computing errors, native code uses float32 here.

It is output data type not inference precision.

Comment (mingmingtasd) [reply examples/asr_sample/utils.js]: Yes, I know that. So should we delete the "error num" in our code or set the threshold to be 1?

Comment (fujunwei) [reply examples/asr_sample/utils.js]: It's better to use the current value for float32 inference precision, and 1 for int8/int16 precision.

Comment (fujunwei) [reply examples/asr_sample/utils.js]: @mingmingtasd Do you have any update on the threshold?

**Comment by fujunwei** [comment]
@miaobin Does the PR fix [the issue](https://github.com/intel/webml-polyfill/issues/1142)?

**Comment by miaobin** [comment]
> @miaobin Does the PR fix [the issue](https://github.com/intel/webml-polyfill/issues/1142)?

No, this PR will not fix it.

File: examples/asr_sample/utils.js (hunk 1 above)
Comment (mingmingtasd) [reply examples/asr_sample/utils.js:265]: It's better to update together after refactoring.

**Comment by fujunwei** [comment]
LGTM, thanks.

**Comment by miaobin** [comment]
> > @miaobin Does the PR fix [the issue](https://github.com/intel/webml-polyfill/issues/1142)?
> 
> No, this PR will not fix it.

Temporarily added ultra_low_power and it will be optimized after sample code refactoring.

**Comment by miaobin** [comment]
@belem We temporarily tested the code and will optimize the code after code refactoring. PTAL.
Thanks.

**Comment by miaobin** [comment]
> Please rename /asr_sample/speech_recognition/g

I have renamed it and updated the README.

**Comment by ibelem** [comment]
@miaobin Please update your example base on #1181

**Comment by miaobin** [comment]
> @miaobin Please update your example based on #1181

OK, I'll update the sample as soon as possible.

**Comment by miaobin** [comment]
Resubmitted the code #1219 and will close it after the new PR merged.

**Comment by miaobin** [comment]
Close it because of #1219 has been merged.

**Comment by huningxin** [comment]
@miaobin , thanks for landing this. However, I cannot find it in [webml examples front page](https://intel.github.io/webml-polyfill/examples). Would you like to add it?

**Comment by ibelem** [comment]
@huningxin I haven't deploy latest code to intel.github.io, i need @Christywl 's test report for @BruceDai 's code refactor.

**Comment by huningxin** [comment]
@ibelem , got it. Looking forward to seeing it.

**Comment by ibelem** [comment]
@huningxin The latest example has been deployed to gh-pages with fix #1229

**Comment by huningxin** [comment]
> @huningxin The latest example has been deployed to gh-pages with fix #1229

Brilliant! Thanks @ibelem @miaobin !

**Pull request closed without merging**